### DISCUSSION TITLE
WWS workload with containerd shim taking more CPU as compared to other Wasm workloads

### DISCUSSION THREAD

**Discussion by divya-mohan0209** [vmware-labs/wasm-workers-server#258] [General]
This is per the discussion on the [Bytecode Alliance Zulip](https://bytecodealliance.zulipchat.com/#narrow/stream/227959-wasmtime-on-k8s/topic/status.20and.20resuscitation/near/404781491) regarding Wasm workers server workloads taking *way more* memory (~38%) as compared to other Wasm workloads from the same example. 

*To do*:
- ~Provide more context about setup~
- ~Provide screencaps from the BA Zulip here, as well.~

**Comment by ereslibre** [comment]
Thank you for creating this discussion @divya-mohan0209! I think is worth taking a look at what is going on.

Are you good if I upload the images here that were uploaded to Zulip?

**Comment by divya-mohan0209** [comment]
@ereslibre : I am sorry I just saw this now :smile: I uploaded the images & have provided more context in the next comment!

wws workload when idle:

<img width="1510" alt="Screenshot-2023-11-29-at-10 41 06AM" src="https://github.com/vmware-labs/wasm-workers-server/assets/21279125/11c7dfaa-de7b-4a04-b5b1-aa7e4cb4517d">

spin workload when idle: 

<img width="1508" alt="Screenshot-2023-11-29-at-10 43 11AM" src="https://github.com/vmware-labs/wasm-workers-server/assets/21279125/5a553190-64a1-4f05-86ad-7e427726ddd2">

**Comment by divya-mohan0209** [comment]
Setup context:
- Operating System: MacOS
- Kubernetes distribution: k3d
- Monitoring software used: Kepler

If you'd like to replicate the setup, you can check out the [demo repo](https://github.com/divya-mohan0209/gdg-cochin-demo).

**Reply by ereslibre** [reply]
Thank you!
